File: sdk/src/main/java/io/antelli/sdk/model/HintList.java
package io.antelli.sdk.model;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.ArrayList;
import java.util.List;

/**
 * Handcrafted by Štěpán Šonský on 02.07.2018.
 */

class HintList implements Parcelable {

    private List<Hint> hints;

    public HintList(){
        hints = new ArrayList();
    }

    public HintList(List<Hint> hints) {
        this.hints = hints;
    }

    public HintList add(Hint hint) {
        this.hints.add(hint);
        return this;
    }

    public HintList addAll(List<Hint> hint) {
        this.hints.addAll(hint);
        return this;
    }

    public List<Hint> getHints() {
        return hints;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeTypedList(this.hints);
    }

    protected HintList(Parcel in) {
        this.hints = in.createTypedArrayList(Hint.CREATOR);
    }

    public static final Creator<HintList> CREATOR = new Creator<HintList>() {
        @Override
        public HintList createFromParcel(Parcel source) {
            return new HintList(source);
        }

        @Override
        public HintList[] newArray(int size) {
            return new HintList[size];
        }
    };
}

File: sdk/src/main/java/io/antelli/sdk/model/HintList.kt
package io.antelli.sdk.model

import android.os.Bundle
import android.os.Parcel
import android.os.Parcelable
import java.util.*

/**
 * Handcrafted by Štěpán Šonský on 02.07.2018.
 */
internal class HintList : Parcelable {
    var hints: MutableList<Hint>
        private set

    constructor() {
        hints = ArrayList()
    }

    constructor(hints: MutableList<Hint>) {
        this.hints = hints
    }

    fun add(hint: Hint): HintList {
        hints.add(hint)
        return this
    }

    fun addAll(hint: List<Hint>): HintList {
        hints.addAll(hint)
        return this
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeTypedList(hints)
    }

    private constructor(parcel: Parcel) {
        hints = parcel.createTypedArrayList(Hint.CREATOR) ?: ArrayList()
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<HintList> = object : Parcelable.Creator<HintList> {
            override fun createFromParcel(source: Parcel): HintList? {
                return HintList(source)
            }

            override fun newArray(size: Int): Array<HintList?> {
                return arrayOfNulls(size)
            }
        }
    }
}